### ISSUE TITLE
Database getting bigger and bigger [Bug?]

### ISSUE THREAD

**Issue by MegaV0lt** [ancestorak/brief#194]
Using brief 2.2.2 and FF 47 shows ma a DB of 148 MB of size in the profile. I already tried to clear unred items, but still 148 MB.
![2016-06-13 14_13_47-fb7pin0c default - freecommander xe](https://cloud.githubusercontent.com/assets/2804301/16007214/632bd71e-3171-11e6-8120-9354716e7f5c.png)

I only have a few bookmarks:
![2016-06-13 14_14_51-brief](https://cloud.githubusercontent.com/assets/2804301/16007243/924cc8d2-3171-11e6-9fbc-393d1a85e0ec.png)

**Comment by Ngamer01** [comment]
Even if you clear your feeds of items within Brief, Brief still needs to set aside some database memory to record feed addresses and the last time they were checked (this is to determine what items are new, what old items were updated, and what items aren't new.  If you need to reduce memory, unsubscribe from feeds you don't need or want.  Also the more feeds you add, the more data Brief will require.  So it's normal that a database memory will grow as you continue to use Brief.

If you want to keep your feeds, but don't want Brief's database to grow out of hand, there are storage options available so that say you can automatically clear items past a certain amount and/or age.  You can even set a check time so you can have Brief automatically check for items as little or as much as you want or need.  Or you can turn off auto-checks and check manually when you need or want.

You can also delete Brief's database when Firefox is closed to generate a new database automatically the next time you use Brief.  You'll just have to put up with possibly redownloading items you've already read though.

**Comment by MegaV0lt** [comment]
I have the option enabled to delete old entrys. I even cklicked "clear all feeds" manualy
![2016-06-19 08_29_25-brief options](https://cloud.githubusercontent.com/assets/2804301/16175716/f8910aec-35f7-11e6-96ca-ec4d7fffd8a1.png)

But still 148MB?

**Comment by Ngamer01** [comment]
It's as I posted:  Even if you clear your feeds of items within Brief, Brief still needs to set aside some database memory to record feed addresses and the last time they were checked (this is to determine what items are new, what old items were updated, and what items aren't new.

Brief will still set aside database memory even if it has to make and keep open spots in the database until new data arrives to fill in said empty spots.

**Comment by tanriol** [comment]
I'll check this, but the database definitely should not take hundreds of MB after being cleared and vacuumed (and Firefox should periodically vacuum it on its own).

**Comment by MichaelEischer** [comment]
tl;dr:
The problems are:
- The trash grows indefinitely until it's emptied manually
- The full-text-search index is not compacted automatically by default
- Vacuum is broken since several years

Clearing all items is similar in effect to emptying the trash: The entries are marked for deletion which happens about once every day. The actual deletion is happening asynchronously!

Long version:
I'm also affected by this bug. Here's what I've found while digging through my backups:
For me the brief.sql file was slowly growing until it required about 120MB. Looking at my later backups showed that I've emptied the trash around that time, afterwards the database was no longer growing for several months. Manually calling vacuum on the databases after emptying the trash using SQLite Manager reduced the size to about 50MB. That is much less than expected when going from 40k entries to 2k. But nevertheless its a significant size reduction.

I've managed to trim the database down to 5MB. So what's causing the remaining size difference? The _entries_text_ table contains the feeds' contents in order to provide the full text search (fts). It is however not compacted when deleting entries. The SQLite manual states that the fts index is by default only garbage-collected during _INSERT_ operations when the amount of blocks grows too far. The implication of this is that deleting entries from the _entries_text_ table doesn't cleanup the index.
Manually calling
`INSERT INTO entries_text(entries_text) VALUES ('merge=200,8');`
a few times cleans up the remainders and thus after a vacuum the database size is down to 5MB. The command could/should be run when purging the deleted entries during idle-daily.
Another option would be to use the automerge command (https://www.sqlite.org/fts3.html#commands).

Taking a look at the storage.vacuum.last.brief.sqlite preference shows (at least for me) that the last vacuum operation was run on December 2014. The breaking change seems to be 70947445c73 which is even older.

**Comment by tanriol** [comment]
The vacuuming process was fixed in v2.3.0, closing.
